### PR TITLE
Emergency cancellation proposal

### PR DESCRIPTION
**Pull request by conectado**
#Tracking #2162

This PR adds the possibility to cancel a proposal that is in any stage (except finalized) by exposing `veto_proposal` as a proposal.

### CODE REVIEW

File: runtime-modules/proposals/engine/src/lib.rs [hunk 1]
```diff
@@ -305,6 +305,9 @@ decl_error! {
         /// Proposal is finalized already
         ProposalFinalized,
 
+        /// Proposal is not pending execution
+        ProposalNotPendingExecution,
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/engine/src/lib.rs]: Do we need this error?

File: runtime/src/proposals_configuration/defaults.rs [hunk 2]
```diff
@@ -347,3 +347,18 @@ pub(crate) fn unlock_blog_post_proposal() -> ProposalParameters<BlockNumber, Bal
         constitutionality: 1,
     }
 }
+
+// TODO: decide on paramaters
```
Comment (shamil-gadelshin) [inline runtime/src/proposals_configuration/defaults.rs]: Please, encode exact parameters.

File: runtime-modules/proposals/engine/src/tests/mod.rs [hunk 3]
```diff
@@ -729,6 +729,98 @@ fn veto_proposal_succeeds() {
...
+        ));
+
+        assert!(!<crate::Proposals<Test>>::contains_key(proposal_id));
+        // internal active proposal counter check
+        assert_eq!(<ActiveProposalCount>::get(), 0);
+    });
+}
+
+#[test]
+fn veto_proposal_succeeds_during_grace_period() {
```
Comment (shamil-gadelshin) [inline runtime-modules/proposals/engine/src/tests/mod.rs:755]: Consider removing this test: proposal_execution_vetoed_successfully_during_the_grace_period.

File: runtime/src/proposals_configuration/defaults.rs [hunk 4]
```diff
@@ -347,3 +347,18 @@ pub(crate) fn unlock_blog_post_proposal() -> ProposalParameters<BlockNumber, Bal
...
+// Proposal parameters for the 'Veto Proposal' proposal
+pub(crate) fn veto_proposal_proposal() -> ProposalParameters<BlockNumber, Balance> {
+    ProposalParameters {
```
Comment (shamil-gadelshin) [inline runtime/src/proposals_configuration/defaults.rs:349]: Please, add the data to the `sample_proposal_parameters.json` and add the `ignored` test for this proposal sample configuration.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good. Left a few comments.

File: runtime/src/proposals_configuration/defaults.rs (hunk 4 above)
Comment (conectado) [reply runtime/src/proposals_configuration/defaults.rs:349]: Do you mean [add this parameter to the json](https://github.com/Joystream/joystream/pull/2178/files#diff-289308fb6811411c14ee716c26aad5fa8b6a9376491ae692d64e4441fea1f715R242) and [this test](https://github.com/Joystream/joystream/pull/2178/files#diff-c03044c26ee60877126fa569ec9c78c6ed7bed8b1f280211cd7a4e3f070d1dc8R235) or am I missing something more?

File: runtime/src/proposals_configuration/tests.rs [hunk 5]
```diff
@@ -228,3 +228,12 @@ fn proposal_parameters_are_initialized_unlock_blog_post_proposal() {
 
     assert_eq!(default_proposal_parameters(), actual_params);
 }
+
+// Enable during the conditional compilation tests.
+#[test]
+#[ignore]
+fn proposal_parameters_are_initialized_veto_proposal_proposal() {
+    let actual_params = super::UnlockBlogPostProposalParameters::get();
```
Comment (shamil-gadelshin) [inline runtime/src/proposals_configuration/tests.rs]: Should be different actual parameters.

File: runtime/src/proposals_configuration/defaults.rs [hunk 6]
```diff
@@ -347,3 +347,17 @@ pub(crate) fn unlock_blog_post_proposal() -> ProposalParameters<BlockNumber, Bal
...
+        voting_period: 72200, // The longest grace period
+        grace_period: 0,
+        approval_quorum_percentage: 75,
+        approval_threshold_percentage: 80,
+        slashing_quorum_percentage: 60,
+        slashing_threshold_percentage: 66,
+        required_stake: Some(1_000_000),
+        constitutionality: 0,
```
Comment (shamil-gadelshin) [inline runtime/src/proposals_configuration/defaults.rs]: Why the constitutionality is set to zero here? Don't we need at least one council approval?

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good. Just one question emerged.

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Comment by conectado** [comment]
Added the [handbook PR](https://github.com/Joystream/handbook/pull/32)

**Pull request merged**